lib/modules/manager/poetry/extract.spec.ts

```diff
@@ -169,6 +170,96 @@ describe('modules/manager/poetry/extract', () => {
       });
     });
 
+    it('parses git dependencies long commit hashs on http urls', async () => {
+      const content = codeBlock`
+        [tool.poetry.dependencies]
+        fastapi = {git = "https://github.com/tiangolo/fastapi.git", rev="6f5aa81c076d22e38afbe7d602db6730e28bc3cc"}
+        werkzeug = ">=0.14"
+      `;
+      const res = (await extractPackageFile(content, filename))!.deps;
+      expect(res[0].depName).toBe('fastapi');
+      expect(res[0].packageName).toBe(
+        'https://github.com/tiangolo/fastapi.git',
+      );
+      expect(res[0].currentValue).toBeUndefined();
+      expect(res[0].currentDigest).toBe(
+        '6f5aa81c076d22e38afbe7d602db6730e28bc3cc',
+      );
+      expect(res[0].replaceString).toBe(
+        '6f5aa81c076d22e38afbe7d602db6730e28bc3cc',
+      );
+      expect(res[0].pinDigests).toBe(true);
+      expect(res[0].skipReason).toBeUndefined();
+      expect(res[0].datasource).toBe(GitRefsDatasource.id);
+      expect(res).toHaveLength(2);
```

Comment:
```suggestion
      expect(res).toMatchObject([])
```
Use `toMatchObject` rather than comparing each evil.